CMakeLists.txt

```diff
@@ -111,6 +111,32 @@ add_subdirectory(features EXCLUDE_FROM_ALL)
 add_subdirectory(cmsis/CMSIS_5/CMSIS/RTOS2 EXCLUDE_FROM_ALL)
 add_subdirectory(cmsis/device/rtos EXCLUDE_FROM_ALL)
 
+
+# We need to generate a "response file" to pass to the C preprocessor because of path length
+# limitations on Windows. We set the response file and bind the path to a global property here.
+# We query this global property when we set the linker script for the MBED_TARGET being built.
+#
+# TODO: Remove this and find a more idiomatic way of passing compile definitions to CPP without
+# using global properties.
+mbed_generate_options_for_linker(${APP_TARGET} LINKER_PREPROCESS_DEFINITIONS)
+set_property(GLOBAL PROPERTY COMPILE_DEFS_RESPONSE_FILE ${LINKER_PREPROCESS_DEFINITIONS})
+
+# This is a temporary workaround to prevent the build from failing for MBED_TARGETS that
+# haven't been converted to build system targets yet.
+# The refactored MBED_TARGETS set the linker script and forward it to the build system as a
+# usage requirement. The 'old' mechanism was to set the linker script on the top level mbed-core
+# target. This was needed because MBED_TARGETS were not registered as buildsystem targets,
+# preventing CMake from working its usage requirements magic and forcing us to set the linker
+# script globally.
+#
+# TODO: Remove when all MBED_TARGETS have been converted to build system targets.
```

Comment:
I am guessing the `if` block would still be needed.
```suggestion
# TODO: Remove `else` block when all MBED_TARGETS have been converted to build system targets.
```

Reply:
I have no idea why the `if` block would be needed. All "mbed targets" will be actual buildsystem targets when we remove this. If someone passes in an incorrect target name we should just fail the build. There's no reason to add a check for that as CMake will check it for us.